services/user/Profiles/plugin/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+#[allow(warnings)]
+mod bindings;
+
+use std::str::FromStr;
+
+use bindings::exports::profiles::plugin::api::Guest as Api;
+use bindings::profiles::plugin::types::Contact;
+use bindings::profiles::plugin::types::Profile as PluginProfile;
+
+use bindings::exports::profiles::plugin::api::File;
+
+use bindings::accounts::plugin::api::get_current_user;
+use bindings::host::common::server as CommonServer;
+use bindings::host::common::types::Error;
+use bindings::transact::plugin::intf::add_action_to_transaction;
+
+use psibase::fracpack::{Pack, Unpack};
+
+use psibase::AccountNumber;
+
+mod errors;
+use errors::ErrorType;
+
+mod contact_table;
+
+struct ProfilesPlugin;
+
+impl Api for ProfilesPlugin {
+    fn set_profile(profile: PluginProfile) {
+        let packed_profile_args = profiles::action_structs::setProfile {
+            display_name: profile.display_name.unwrap_or_default(),
+            bio: profile.bio.unwrap_or_default(),
+            avatar: profile.avatar,
+        }
+        .packed();
+
+        add_action_to_transaction("setProfile", &packed_profile_args).unwrap();
+    }
+
+    fn upload_avatar(file: File, compression_quality: u8) -> Result<(), Error> {
```

Comment:
You can make the choice for the compression quality inside this plugin and then you don't need to pass it along to the GUI. I'd probably just use max compression (11).